mat64/dense.go

```diff
@@ -427,3 +429,63 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
```

Comment:
Since we intend this to be cross-language and cross-program, I think a comment documenting how the matrix is marshaled is in order.